### PR TITLE
Remove Event from API, favoring Observable functions

### PR DESCRIPTION
**Pull request by eliperkins**
Depends on #3 

Event was a bit poorly named and it's API a bit misguided. Event
contained duplication of data, namely around `value` and the way changes
to this value were handled. Additionally, adding functionality on by
giving new properties to Observable created an API to getting these
values out that wasn't as straight forward as could be.

This removes the functionality given by Event back into Observable
(namely, `subscribe` and the handling of its invocations).

Also, `new` has been removed and `newOld` replaced with `func
changesets()` that should mirror it's functionality. The difference in
`changesets()` comes in that `NewOldEvent` is no longer needed, since we
can just modify the generic type that is constraining `Observable` to
reflect the modification of the parent `Observable`.

Finally, `Observable` has been modified to be a class instead of a
struct. This is due to the inherent nature of its statefulness, around
maintaining its subscribers, as well as it's `value`.

A bit of inspiration here from ReactiveCocoa as well.

### CODE REVIEW

File: Notice/Observable.swift [hunk 1]
```diff
@@ -9,12 +9,13 @@ public struct Observable<T> {
...
         didSet {
             dispatch_sync(eventQueue) {
-                self.new.invoke(self.value)
-                self.newOld.invoke((old: oldValue, new: self.value))
+                self.invoke(self.value)
             }
         }
     }
 
+    public var subscriptions = Set<Subscription<T -> Void>>()
```
Comment (hyperspacemark) [inline Notice/Observable.swift]: What are your feels on a typealias for `T -> Void`?

Comment (eliperkins) [inline Notice/Observable.swift]: They feel pretty good. Contemplated doing that before, but bailed. You mentioning that now leads me to believe it should be done.

Comment (eliperkins) [inline Notice/Observable.swift]: Done in 9e64735

**Comment by lionel-alves** [comment]
:green_apple:

**Pull request merged**